sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java

```diff
@@ -120,4 +123,22 @@ void fillCosmosDiagnostics(ObjectNode parentNode, StringBuilder stringBuilder) {
     void setFeedResponseDiagnostics(FeedResponseDiagnostics feedResponseDiagnostics) {
         this.feedResponseDiagnostics = feedResponseDiagnostics;
     }
+
+    private AtomicBoolean isDiagnosticsCapturedInPagedFlux(){
```

Comment:
could you please high level explain what the purpose of this method is?

Reply:
No we are wrapping the CosmosPageFlux twice , so to avoid duplicate capture of diagnostics in client telemetry , tracer(future work item)